### PR TITLE
feat(RUN-971): Add page listing Execution errors

### PR DESCRIPTION
**Pull request by adambratschikaye**
Add a doc page which will be filled in with information about various
execution errors. Links to this page will be provided with each errors
returned by ICP.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -6,5 +6,6 @@ docs/developer-docs/build/agents @dfinity/dx
 docs/developer-docs/build/cdks @dfinity/dx
 docs/developer-docs/integrations/rosetta @dfinity/finint
 docs/developer-docs/security @dfinity/product-security
+docs/references/execution-errors.mdx @dfinity/runtime @dfinity/execution
```
Comment (jessiemongeon1) [inline .github/CODEOWNERS:9]: "Unknown owner: make sure that the team @dfinity/runtime and @dfinity/execution exist, are publicly visible, and has write access to the repository".

File: docs/references/execution-errors.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Execution Errors
```
Comment (jessiemongeon1) [inline docs/references/execution-errors.mdx:3]:
```suggestion
# Execution errors
```

File: docs/references/execution-errors.mdx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+<MarkdownChipRow labels={["Reference"]} />
+
+## Overview
+A list of possible errors returned when executing canisters.
+
+## Errors
+
+### Method Not found
```
Comment (jessiemongeon1) [inline docs/references/execution-errors.mdx:12]:
```suggestion
### Method not found
```

File: docs/references/execution-errors.mdx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+A canister was called with a method name not exported by that canister.
+
+An example of this error is:
+```
+  Error from Canister xxx-xxx: Canister has no update method 'foobar'.
+```
+
+To fix this error, consider
```
Comment (jessiemongeon1) [inline docs/references/execution-errors.mdx:20]:
```suggestion
To fix this error, consider:
```

File: docs/references/execution-errors.mdx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+1. Checking that the method name exacly matches the name exported by the callee.
+1. Checking that the method type (update, query, etc...) is the same as the type of the exported method.
+1. The canister code of the callee has not been modified to a version that depricates the method.
```
Comment (jessiemongeon1) [inline docs/references/execution-errors.mdx:24]:
```suggestion
- Checking that the method name exactly matches the name exported by the callee.
- Checking that the method type (update, query, etc...) is the same as the type of the exported method.
- The canister code of the callee has not been modified to a version that depreciates the method.
```

**Review by jessiemongeon1 — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (adambratschikaye) [reply .github/CODEOWNERS:9]: Any idea who has control of that? The teams seem to exist and be visible based on [this search](https://github.com/orgs/dfinity/teams?page=1&query=+visibility%3Avisible), so I guess the issue is that they don't have write access to the rep?

File: docs/references/execution-errors.mdx (hunk 5 above)
Comment (adambratschikaye) [reply docs/references/execution-errors.mdx]: I fixed "exactly" and the bullets, but I think "depricates" is actually correct.

Comment (jessiemongeon1) [reply docs/references/execution-errors.mdx]: "depricates" is not a dictionary recognized word: https://www.merriam-webster.com/dictionary/depricates

Comment (jessiemongeon1) [reply docs/references/execution-errors.mdx]: However depreciate is also incorrect; it should be 'deprecate' if you mean the method is no longer supported.

Comment (adambratschikaye) [reply docs/references/execution-errors.mdx]: I believe "depricates" is in Merriam Webster, it's just a conjugation of "depricate": https://www.merriam-webster.com/dictionary/deprecate

Comment (jessiemongeon1) [reply docs/references/execution-errors.mdx]: Yes, it should be `deprecate`, with an `e` not an `i`.

Comment (adambratschikaye) [reply docs/references/execution-errors.mdx]: oh, yeah of course.

File: .github/CODEOWNERS (hunk 1 above)
Comment (jessiemongeon1) [reply .github/CODEOWNERS:9]: Yea it looks like they need write access to the repo. I'm not sure who can provide that - IT maybe?

**Pull request merged**